internal/kafka/internal/workers/cluster_mgrs/clusters_mgr.go

```diff
@@ -777,9 +777,15 @@ func (c *ClusterManager) buildResourceSet(cluster api.Cluster) types.ResourceSet
 		c.buildObservabilityCatalogSourceResource(),
 		c.buildObservabilityOperatorGroupResource(),
 		c.buildObservabilitySubscriptionResource(),
-		c.buildObservabilityRemoteWriteServiceAccountCredential(&cluster),
 	)
 
+	// if no oidc configuration is provided, or if it is disabled, use the data plane service account
+	if !(c.ObservabilityConfiguration.DataPlaneObservabilityConfig.Enabled && c.ObservabilityConfiguration.DataPlaneObservabilityConfig.HasOIDCConfiguration()) {
```

Comment:
I find the way this condition is written is confusing.
The comment describes the more natural way that it is described:
if !enabled || !hasoidconfiguration would be more understandable.